src/pushsource/_impl/model/ami.py

```diff
@@ -129,6 +129,11 @@ class AmiPushItem(VMIPushItem):
     )
     """``True`` if the image supports Elastic Network Adapter (ENA)."""
 
+    uefi_support = attr.ib(
+        type=bool, default=None, validator=optional(instance_of(bool))
+    )
+    """``True`` if the image supports UEFI boot."""
+
```

Comment:
@amcmahon-rh It is not such simple boolean value.
Image builder provides additional metadata in koji build e.g.
```
 'extra': {'typeinfo': {'image': {'rhel-azure-9.3-20230823.14.aarch64.vhd.xz': {'arch': 'aarch64',
                                                                                'boot_mode': 'uefi'},
                                  'rhel-azure-9.3-20230823.14.x86_64.vhd.xz': {'arch': 'x86_64',
                                                                               'boot_mode': 'hybrid'}}}},
```

aarch64 obviously has to use uefi. But with x86_64 it should try uefi but it can do fallback
in AWS terminology, it is 3 state: "uefi-preferred", "legacy-bios" and "uefi"
* https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ami-boot-mode.html
And one cannot achieve that with boolean in cloudimg
* https://github.com/amcmahon-rh/cloudimg/blob/1ae154580abbeb5ed555d443a2c9bcd61ed1ef70/cloudimg/aws.py#L53

Reply:
Hey @lslebodn, I don't quite understand, could you elaborate please? 

I understand that AWS requires specific string values. The idea of using a boolean was to keep the value abstract up to the point where we actually interact with AWS, rather than passing string values around different libraries. 

Would this be resolved by having a second boolean? One for uefi and one for bios support?